main/main.go

```diff
@@ -254,76 +265,178 @@ func main() {
 	)
 }
 
-// runRelayer creates a relayer instance for a subnet. It listens for warp messages on that subnet, and handles delivery to the destination
-func runRelayer(
+// runListener creates a Listener instance and the ApplicationRelayers for a subnet.
+// The Listener listens for warp messages on that subnet, and the ApplicationRelayers handle delivery to the destination
+func runListener(
 	ctx context.Context,
 	logger logging.Logger,
 	metrics *relayer.ApplicationRelayerMetrics,
 	db database.RelayerDatabase,
 	ticker *utils.Ticker,
-	sourceSubnetInfo config.SourceBlockchain,
+	sourceBlockchain config.SourceBlockchain,
 	network *peers.AppRequestNetwork,
-	responseChan chan message.InboundMessage,
 	destinationClients map[ids.ID]vms.DestinationClient,
 	messageCreator message.Creator,
 	relayerHealth *atomic.Bool,
-	manualWarpMessages []*relayerTypes.WarpLogInfo,
+	manualWarpMessages []*relayerTypes.WarpMessageInfo,
 	cfg *config.Config,
 ) error {
-	logger.Info(
-		"Creating relayer",
-		zap.String("originBlockchainID", sourceSubnetInfo.BlockchainID),
+	// Dial the eth client
+	ethClient, err := ethclient.DialWithConfig(
+		context.Background(),
+		sourceBlockchain.RPCEndpoint.BaseURL,
+		sourceBlockchain.RPCEndpoint.HTTPHeaders,
+		sourceBlockchain.RPCEndpoint.QueryParams,
 	)
+	if err != nil {
+		logger.Error(
+			"Failed to connect to node via RPC",
+			zap.String("blockchainID", sourceBlockchain.BlockchainID),
+			zap.Error(err),
+		)
+		return err
+	}
 
-	listener, err := relayer.NewListener(
+	// Create the ApplicationRelayers
+	applicationRelayers, minHeight, err := createApplicationRelayers(
```

Comment:
May want to consider moving the creation of application relayers out of `runListener` such that the application relayer instances can be used/passed else where if needed. 

For instance, in the case of an API to manually trigger the relaying of a message, I think it would want to route the message relay request to the respective application relayer if applicable. This way, the reference to the application relayers isn't isolated to the listeners.